### ISSUE TITLE
Add `.name.suffix`

### ISSUE THREAD

**Issue by MarcoGorelli** [narwhals-dev/narwhals#130]
From a request  [here](https://github.com/koaning/scikit-lego/pull/661#discussion_r1596527264)

**Comment by FBruzzesi** [comment]
I have implementation for most methods, however there might be an issue: `PandasSeries` already defines a [`name` property](https://github.com/narwhals-dev/narwhals/blob/d6b0ca5231f18f2cdb89ca6b9cd323170c07793c/narwhals/_pandas_like/series.py#L121)
which makes impossible to register the namespace 😕

**Comment by MarcoGorelli** [comment]
wow, I'm impressed!

🤔 doesn't polars.Series also have that? https://github.com/pola-rs/polars/blob/11fe9d8f223d50b5e3284b924bb969049b0be6c6/py-polars/polars/series/series.py#L589-L590

I think `.name.suffix` should only go on expressions, not series, right?

**Comment by FBruzzesi** [comment]
Oh I see, instead of defining the methods on series _and then_ register them, it is enough to define them in the `PandasExpr`

**Comment by MarcoGorelli** [comment]
yeah, exactly! the register function is just a trick which allows you to reuse the Series implementation. `register_expression_call` isn't a good function name to be fair...`reuse_series_implementation` might be better?

**Comment by FBruzzesi** [comment]
I am a bit uncertain of a detail, I will open a PR so that we can discuss with code on hand :)
